internal/pkg/cli/local_run.go

```diff
@@ -406,6 +443,141 @@ func (o *localRunOpts) runContainers(ctx context.Context, imageInfoList []clidep
 	return nil
 }
 
+type envVarValue struct {
+	Value  string
+	Secret bool
+}
+
+// getEnvVars uses env overrides passed by flags and environment variables/secrets
+// specified in the Task Definition to return a set of environment varibles for each
+// continer defined in the TaskDefinition. The returned map is a map of container names,
+// each of which contains a mapping of key->envVarValue, which defines if the variable is a secret or not.
+func (o *localRunOpts) getEnvVars(ctx context.Context, taskDef *awsecs.TaskDefinition) (map[string]map[string]envVarValue, error) {
+	envVars := make(map[string]map[string]envVarValue)
+	for _, ctr := range taskDef.ContainerDefinitions {
+		envVars[aws.StringValue(ctr.Name)] = make(map[string]envVarValue)
+	}
+
+	if err := o.fillEnvOverrides(envVars); err != nil {
+		return nil, fmt.Errorf("parse env overrides: %w", err)
+	}
+
+	for _, e := range taskDef.EnvironmentVariables() {
+		// only set if it wasn't overridden
+		if _, ok := envVars[e.Container][e.Name]; !ok {
+			envVars[e.Container][e.Name] = envVarValue{
+				Value: e.Value,
+			}
+		}
+	}
+
+	if err := o.fillSecrets(ctx, envVars, taskDef); err != nil {
+		return nil, fmt.Errorf("get secrets: %w", err)
+	}
+	return envVars, nil
+}
+
+// fillEnvOverrides parses environment variable overrides passed via flag.
+// The expected format of the flag values is KEY=VALUE, with an optional container name
+// in the format of [containerName]:KEY=VALUE. If the container name is omitted,
+// the environment variable override is applied to all containers in the task definition.
+func (o *localRunOpts) fillEnvOverrides(envVars map[string]map[string]envVarValue) error {
+	for k, v := range o.envOverrides {
```

Comment:
Same feedback as port override - a new type could help from readability/reusability's perspective 💭

Reply:
yeah that's a good idea - i'll see what we can do, the main issue is that these are filled by the pflag library, and custom types are kinda annoying there 🙃

Reply:
it's basically a special string type i think pflag can do it.

Reply:
ok i made some small changes! let me know what you think. if you think it's clearer, i can make a [custom type](https://pkg.go.dev/github.com/spf13/pflag?utm_source=godoc#Value) to work with the pflag library. here's an example: https://github.com/spf13/pflag/blob/v1.0.5/string_to_string.go#L11-L77

Reply:
After a second thought i think we should probably implement this since most logic is already here it'll help error out earlier if they put any invalid env override (it requires the input to be in some certain format) 🙇

Reply:
ok, done! i added a custom type to parse the `portOverrides` flag in to!